src/components/ItemModal.ts

```diff
@@ -27,6 +27,38 @@ export class ItemModal extends Modal {
     return this.isValidName ? "" : "This is an invalid name. Name's can't include /, \\, :, or .";
   }
 
+  checkIfDatesEmpty(date_setting: string ){
+    if(date_setting == 'Do Date')
+        return this.successPlanItem.do_date != "" ? this.successPlanItem.do_date._i : "";
+    else if(date_setting == 'Due Date')
+        return this.successPlanItem.due_date != "" ? this.successPlanItem.due_date._i : "";
+    else 
+        return this.successPlanItem.closing_date != "" ? this.successPlanItem.closing_date._i : "";
+  }
+
+  getSuccessItemProperty(date_setting: string){
+    return date_setting == 'Do Date' ? this.successPlanItem.do_date : (date_setting == 'Due Date' ? this.successPlanItem.due_date : this.successPlanItem.closing_date);
```

Comment:
Generally speaking, you never want to nest ternary statements. It makes the code hard to read.